.github/workflows/build-test-distribute.yaml

```diff
@@ -129,3 +134,96 @@ jobs:
+        env:
+          SBOM_DOWNLOAD_PATH: "${{ github.workspace }}/security-assets/sboms"
+          SPDX_SBOM_PATTERN: "*sbom.spdx.json"
+          CYCLONEDX_SBOM_PATTERN: "*sbom.cyclonedx.json"
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # uses: actions/download-artifact@v4
+          # id: collect_sbom
+          # with:
+          #   path: ${{ github.workspace }}/security-assets/sboms
+          #   pattern: "*sbom.{cyclonedx,spdx}.json"
+          #   merge-multiple: true # When set, all matched zipped artifacts are extracted into the specified path
+      - name: "Download binary artifact provenance"
+        if: ${{ needs.provenance.result == 'success' && github.ref_type == 'tag' }}
+        id: collect_provenance
+        run: |-
+          echo "PROVENANCE_DOWNLOAD_PATH=${{env.PROVENANCE_DOWNLOAD_PATH}}" >> $GITHUB_OUTPUT
+          gh run download ${{ github.run_id }} -D ${{ env.PROVENANCE_DOWNLOAD_PATH }} -n ${{ env.BINARY_PROVENANCE_ARTIFACT }} --repo ${{ github.repository }}
+        env:
+          PROVENANCE_DOWNLOAD_PATH: "${{ github.workspace }}/security-assets/provenance"
+          BINARY_PROVENANCE_ARTIFACT: "${{ github.event.repository.name }}.intoto.jsonl"
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # uses: actions/download-artifact@v4
+          # with:
+          #   path: ${{ github.workspace }}/security-assets/provenance
+          #   pattern: ${{ github.event.repository.name }}.intoto.jsonl
+          #   merge-multiple: true # When set, all matched zipped artifacts are extracted into the specified path
+      - name: "check sbom asset files existence"
+        uses: andstor/file-existence-action@v3
+        id: check_sbom_assets
+        if: ${{ fromJSON(needs.check.outputs.BUILD) && (needs.check.result == 'success' || needs.build_publish.result == 'success') }}
+        with:
+          files: ${{ steps.collect_sbom.outputs.SBOM_DOWNLOAD_PATH }}
+          fail: true
+      - name: "check provenance asset files existence"
+        uses: andstor/file-existence-action@v3
+        if: ${{ needs.provenance.result == 'success' && github.ref_type == 'tag' }}
+        id: check_provenance_assets
+        with:
+          files: ${{ steps.collect_sbom.outputs.PROVENANCE_DOWNLOAD_PATH }}
+          fail: true
+      - name: "prepare sbom package" # Zip all SBOM assets for artifact types (Images, Repository scanning) produced in the jobs
+        id: prepare_sbom_metadata
+        if: ${{ steps.check_sbom_assets.outputs.file_exists == 'true' }}
+        run: |-
+          SBOM_PACKAGE_NAME="${{github.repository}}-sbom"
+          echo "SBOM_PACKAGE_NAME=${SBOM_PACKAGE_NAME}" >> $GITHUB_OUTPUT
+          zip -rj ${SBOM_PACKAGE_NAME} ${{steps.collect_sbom.outputs.download-path}} -i '*sbom.spdx.json' '*sbom.cyclonedx.json'
```

Comment:
Do we want to use a zip? a tar.gz is usually more common no?

Reply:
Using TAR